### PR TITLE
Feature/#16 real support for junit new

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**

### CODE REVIEW

File: accurest-gradle-plugin/src/test/resources/functionalTest/sampleProject/fraudDetectionService/src/test/java/com/blogspot/toomuchcoding/MvcTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+public class MvcTest {
+
+	@Before
+	public void setup() {
+		RestAssuredMockMvc.standaloneSetup(new com.blogspot.toomuchcoding.frauddetection.FraudDetectionController());
+	}
+
+	public void assertThatRejectionReasonIsNull(Object rejectionReason) {
+		assert rejectionReason== null;
```
Comment (marcingrzejszczak) [inline accurest-gradle-plugin/src/test/resources/functionalTest/sampleProject/fraudDetectionService/src/test/java/com/blogspot/toomuchcoding/MvcTest.java]: space is missing after `rejectionReason`

Comment (jkubrynski) [inline accurest-gradle-plugin/src/test/resources/functionalTest/sampleProject/fraudDetectionService/src/test/java/com/blogspot/toomuchcoding/MvcTest.java]: :+1:

Comment (jkubrynski) [inline accurest-gradle-plugin/src/test/resources/functionalTest/sampleProject/fraudDetectionService/src/test/java/com/blogspot/toomuchcoding/MvcTest.java]: except this comment it looks good for me :)

File: accurest-core/src/main/groovy/io/codearte/accurest/SingleTestGenerator.groovy [hunk 2]
```diff
@@ -15,7 +15,8 @@ import static io.codearte.accurest.util.NamesUtil.capitalize
 @Slf4j
 class SingleTestGenerator {
 
-	private static final String JSON_ASSERT_STATIC_IMPORT = 'com.blogspot.toomuchcoding.jsonassert.JsonAssertion.assertThat'
+	private static
+	final String JSON_ASSERT_STATIC_IMPORT = 'com.blogspot.toomuchcoding.jsonassert.JsonAssertion.assertThatJson'
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/SingleTestGenerator.groovy]: Why two lines?

File: accurest-core/src/main/groovy/io/codearte/accurest/builder/JaxRsClientJUnitMethodBodyBuilder.groovy [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	@Override
+	protected void when(BlockBuilder bb) {
+		bb.addLine("Response response = webTarget")
+		bb.indent()
+
+		appendUrlPathAndQueryParameters(bb)
+		appendRequestWithRequiredResponseContentType(bb)
+		appendHeaders(bb)
+		appendMethodAndBody(bb)
+		bb.addAtTheEnd(';')
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/builder/JaxRsClientJUnitMethodBodyBuilder.groovy]: There is TestFramework.lineSuffix which you should use

Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/builder/JaxRsClientJUnitMethodBodyBuilder.groovy]: There is TestFramework.lineSuffix which you should use

File: accurest-core/src/main/groovy/io/codearte/accurest/SingleTestGenerator.groovy (hunk 2 above)
Comment (OlgaMaciaszek) [inline accurest-core/src/main/groovy/io/codearte/accurest/SingleTestGenerator.groovy]: Autoformatter :). Fixing

File: accurest-core/src/main/groovy/io/codearte/accurest/builder/JaxRsClientSpockMethodBodyBuilder.groovy [hunk 4]
```diff
@@ -15,6 +16,9 @@ class JaxRsClientSpockMethodBodyBuilder extends SpockMethodBodyBuilder {
 	}
 
 	@Override
+	protected void given(BlockBuilder bb) {}
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/builder/JaxRsClientSpockMethodBodyBuilder.groovy:19]: What's a diffrernce between this and givenBlock?

File: accurest-core/src/main/groovy/io/codearte/accurest/builder/MethodBodyBuilder.groovy [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+	}
+
+	protected void appendJsonPath(BlockBuilder blockBuilder, String json) {
+		blockBuilder.addLine(("DocumentContext parsedJson = JsonPath.parse($json)"))
+		addColonIfRequired(blockBuilder)
+	}
+
+	protected void processText(BlockBuilder blockBuilder, String property, String value) {
+		if (value.startsWith('$')) {
+			value = value.substring(1).replaceAll('\\$value', "responseBody$property")
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/builder/MethodBodyBuilder.groovy]: It looks really unreadable -maybe extract it to a utility method `stripFirstChar` if it is not exising already in some StringUtils

File: accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/SampleProjectSpec.groovy [hunk 6]
```diff
@@ -2,20 +2,48 @@ package io.codearte.accurest.plugin
...
+import java.nio.file.Files
+import java.nio.file.Path
+
+import static java.nio.charset.StandardCharsets.UTF_8
+import static java.nio.charset.StandardCharsets.UTF_8
 
 @Stepwise
 class SampleProjectSpec extends IntegrationSpec {
 
+	public static final String SPOCK = "targetFramework = 'Spock'"
```
Comment (jkubrynski) [inline accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/SampleProjectSpec.groovy]: Maybe move those fields to an abstract AccurestIntegrationSpec?

File: accurest-core/src/main/groovy/io/codearte/accurest/builder/JaxRsClientSpockMethodBodyBuilder.groovy (hunk 4 above)
Comment (OlgaMaciaszek) [inline accurest-core/src/main/groovy/io/codearte/accurest/builder/JaxRsClientSpockMethodBodyBuilder.groovy:19]: In this class it's same, but they they override two different methods with two different implementations in the base class. Basically, I want to suppress the base class code for JaxRs.

File: accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/SampleProjectSpec.groovy (hunk 6 above)
Comment (OlgaMaciaszek) [inline accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/SampleProjectSpec.groovy]: Yes, that's a good idea. Adding.

**Comment by jkubrynski** [comment]
LGTM :) Go!

**Pull request merged**